src/arbitrage.py

```diff
@@ -0,0 +1,38 @@
+class Query (object):
+
+    def __init__(self, limit=0, stock_names=None, api=None):
+        self.limit = limit
+        self.names = stock_names
+        self.api = api
+
+    def pull(self, stock):
+        """Connects to API to pull data for a given stock"""
+        return
+
+    def check(self, expiry_dates=None, diff_costs=None, underlying_cost=None):
+        """Checks to see if there exists an expiry date for which
+        the arbitrage formula for options holds"""
+        result = False
+        for expiry in expiry_dates:
+            diff_cost = diff_costs[expiry]
+            if diff_cost / underlying_cost > self.limit:
+                result = True
+        return result
+
+    def find_opportunities(self):
+        """Prints the stocks that satisfy the arbitrage opportunity requirements"""
+        for name in self.names:
+            data = self.pull(name)
+            expiry_dates = data.get_dates(name)
+            underlying_cost = data.get_underlying(name)
+            diff_costs = data.get_diff_costs(name)
```

Comment:
Why would data give you diff costs? It would only give you the spot price at each time, you would have to get the difference yourself.

Reply:
fixed